editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/adopters/configurators/MavenTemplateConfigurator.java

```diff
@@ -111,29 +108,28 @@ private void waitAllJobsComplete(IProgressMonitor monitor) {
 	 * @return	true on success, otherwise false
 	 */
 	protected boolean configurePomCamelVersion(IProject project, NewProjectMetaData projectMetaData, IProgressMonitor monitor) {
-		if (monitor == null) {
-			monitor = new NullProgressMonitor();
-		}
+		SubMonitor subMonitor = SubMonitor.convert(monitor,Messages.MavenTemplateConfigurator_AdaptingprojectToCamelVersionMonitorMessage, 5);
 		try {
-			monitor.beginTask("Adapting the project to the new Camel version...", IProgressMonitor.UNKNOWN);
-			File pomFile = new File(project.getFile("pom.xml").getLocation().toOSString());
+			File pomFile = new File(project.getFile("pom.xml").getLocation().toOSString()); //$NON-NLS-1$
 			Model m2m = MavenPlugin.getMaven().readModel(pomFile);
-
+			subMonitor.worked(1);
 			final String camelVersion = projectMetaData.getCamelVersion();
 			if (m2m.getDependencyManagement() != null) {
 				MavenUtils.updateCamelVersionDependencies(m2m.getDependencyManagement().getDependencies(), camelVersion);
 			}
+			subMonitor.worked(1);
 			MavenUtils.updateCamelVersionDependencies(m2m.getDependencies(), camelVersion);
 			if (m2m.getBuild().getPluginManagement() != null) {
 				MavenUtils.updateCamelVersionPlugins(m2m.getBuild().getPluginManagement().getPlugins(), camelVersion);
 			}
+			subMonitor.worked(1);
 			MavenUtils.updateCamelVersionPlugins(m2m.getBuild().getPlugins(), camelVersion);
-
+			subMonitor.worked(1);
 			OutputStream os = new BufferedOutputStream(new FileOutputStream(pomFile));
 		    MavenPlugin.getMaven().writeModel(m2m, os);
-			IFile pomIFile2 = project.getProject().getFile("pom.xml");
+			IFile pomIFile2 = project.getProject().getFile("pom.xml"); //$NON-NLS-1$
 			if (pomIFile2 != null) {
-				pomIFile2.refreshLocal(IResource.DEPTH_INFINITE, monitor);
+				pomIFile2.refreshLocal(IResource.DEPTH_INFINITE, subMonitor.newChild(1));
```

Comment:
A question rather than a comment: Would the monitor handle things correctly if you only allocated 4 in the convert, but this if statement ended up adding a fifth segment?